src/similarity.rs

```diff
@@ -54,6 +54,15 @@ pub trait Analogy {
         word3: &str,
         limit: usize,
     ) -> Option<Vec<WordSimilarity>>;
+
+    fn analogy_masked(
```

Comment:
The `analogy` method above could now provide a default implementation that calls `analogy_by` the `analogy` implementation in the `impl` block can then be removed.